From ae13ca20e4514f02cb560cbdec745aad4c2980c0 Mon Sep 17 00:00:00 2001
From: Vanessa Maegima <vanessa.maegima@foundries.io>
Date: Fri, 9 Jul 2021 16:12:53 -0300
Subject: [PATCH] imx8mm_var_dart: define CONFIG_SYS_BOOTM_LEN

Signed-off-by: Vanessa Maegima <vanessa.maegima@foundries.io>
---
 include/configs/imx8mm_var_dart.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/imx8mm_var_dart.h b/include/configs/imx8mm_var_dart.h
index ade6cb42f5..96e7ec0d49 100644
--- a/include/configs/imx8mm_var_dart.h
+++ b/include/configs/imx8mm_var_dart.h
@@ -214,6 +214,7 @@
 #define DEFAULT_SDRAM_SIZE		(512 * SZ_1M) /* 512MB Minimum DDR4, see get_dram_size */
 #define VAR_EEPROM_DRAM_START          (CONFIG_SYS_MEMTEST_START + \
 					(DEFAULT_SDRAM_SIZE >> 1))
+#define CONFIG_SYS_BOOTM_LEN		SZ_64M
 
 #define CONFIG_SYS_MEMTEST_START	PHYS_SDRAM
 #define CONFIG_SYS_MEMTEST_END		(CONFIG_SYS_MEMTEST_START + (DEFAULT_SDRAM_SIZE >> 1))
-- 
2.25.1

