From a157957c4eaacbd16428a7c197decdac7f1cea4a Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@foundries.io>
Date: Thu, 28 Jan 2021 17:34:22 +0200
Subject: [PATCH 082/322] [FIO internal] usb hub: don't use env if it's not
 enabled

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
---
 common/usb_hub.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/common/usb_hub.c b/common/usb_hub.c
index 3d856e7de79..a3aef39162a 100644
--- a/common/usb_hub.c
+++ b/common/usb_hub.c
@@ -165,8 +165,9 @@ static void usb_hub_power_on(struct usb_hub_device *hub)
 	int i;
 	struct usb_device *dev;
 	unsigned pgood_delay = hub->desc.bPwrOn2PwrGood * 2;
+#if !defined(CONFIG_SPL_BUILD) && !defined(CONFIG_SPL_ENV_SUPPORT)
 	const char *env;
-
+#endif
 	dev = hub->pusb_dev;
 
 	debug("enabling power on all ports\n");
@@ -190,10 +191,12 @@ static void usb_hub_power_on(struct usb_hub_device *hub)
 	 * but allow this time to be increased via env variable as some
 	 * devices break the spec and require longer warm-up times
 	 */
+#if !defined(CONFIG_SPL_BUILD) && !defined(CONFIG_SPL_ENV_SUPPORT)
 	env = env_get("usb_pgood_delay");
 	if (env)
 		pgood_delay = max(pgood_delay,
 			          (unsigned)simple_strtol(env, NULL, 0));
+#endif
 	debug("pgood_delay=%dms\n", pgood_delay);
 
 	/*
-- 
2.34.1

